lib/galaxy/schema/schema.py

```diff
@@ -1410,8 +1414,40 @@ class JobExportHistoryArchiveModel(Model):
     )
 
 
+class ExportHistoryRequestMetadata(Model):
+    history_id: EncodedDatabaseIdField
+    user_id: Optional[EncodedDatabaseIdField]
+    payload: WriteStoreToPayload
+
+
+class ExportHistoryResultMetadata(Model):
+    published: bool
+    # TODO: figure out what else is needed
+
+
+class ExportHistoryMetadata(Model):
+    request_data: ExportHistoryRequestMetadata
+    result_data: ExportHistoryResultMetadata
+
+
+class TaskExportHistoryArchiveModel(ExportHistoryArchiveBaseModel):
+    type: Literal["task"]
+    task_uuid: UUID4 = Field(
+        ...,
+        title="Task ID",
+        description="The identifier of the celery task processing the export.",
+    )
+    create_time: datetime = CreateTimeField
+    export_metadata: Optional[ExportHistoryMetadata]
+
+
+AnyExportHistoryArchiveModel = Annotated[
+    Union[JobExportHistoryArchiveModel, TaskExportHistoryArchiveModel], Field(discriminator="type")
+]
+
+
 class JobExportHistoryArchiveCollection(Model):
-    __root__: List[JobExportHistoryArchiveModel]
+    __root__: List[AnyExportHistoryArchiveModel]
```

Comment:
I'm trying to decide if returning both types of "views" (Job or Task) of a history export will be too much of a breaking change for the API...

Will it be more desirable to create a dedicated route for the task-based exports (like `api/histories/{id}/exports/tasks`) and leave the current `api/histories/{id}/exports` just for the job-based one so the API doesn't change that much?
OTOH I feel like it will be a bit harder to maintain in the client later if you need to request both endpoints to know how the history was exported... :thinking:

Reply:
I guess the question boils down to... what is the best approach to associate export metadata to an exported "object" (not necessarily only histories) so we can track when this object was exported and where (for recovery)?

I was hoping to use `api/histories/{id}/exports` for the particular case of histories because it makes a lot of sense semantically. But maybe we should use a more general API independent of the object type?

Reply:
Following up with the discussion yesterday in the backend meeting about this...

1 - Is it a better idea to create a new `GET /api/histories/{id}/store_exports` endpoint instead of trying to reuse `GET /api/histories/{id}/exports`? this **store_exports** endpoint will have an equivalent for all objects that support the **write_store** and **from_store** endpoints (like invocations)

2 - Assuming we want to use case (1). How do we want to persist the export metadata for all these objects (histories, invocations, etc)? Creating a new "store_export_association" table that tracks all the exports (excluding simple downloads) for all types of exported objects? 

3 - Do we want to deprecate the job-based export for histories and use only the task-based one at some point? Meaning we should update the current history export UI to align it more with #14606?

Reply:
> Creating a new "store_export_association" table that tracks all the exports (excluding simple downloads) for all types of exported objects?

I think that's a solid yes.

> this **store_exports** endpoint will have an equivalent for all objects that support the **write_store** and **from_store** endpoints (like invocations)

You mean we'll also have `/api/invocations/{id}/store_exports` ? This sounds good to me, my one concern is that `store` feels like an implementation detail consumers wouldn't need to know about. Is there any chance we can gloss over the differences and keep a backwards compatible API at `/api/histories/{id}/exports ?

> Do we want to deprecate the job-based export for histories and use only the task-based one at some point? Meaning we should update the current history export UI to align it more with #14606?

Yes IMO.

Reply:
Thank you @mvdbeek, that helps a lot!

> You mean we'll also have /api/invocations/{id}/store_exports

Yes, that is what I meant :)

> Is there any chance we can gloss over the differences and keep a backwards compatible API at /api/histories/{id}/exports

The only solution that pops into my mind is using a custom content-type for returning just the task-based exports and fallback to the job-based one with the default content-type. Will that be OK in this case?

> Yes IMO. That would close https://github.com/galaxyproject/galaxy/issues/14511

Oh! great thanks! I lost track of this one :)

Reply:
I'm happy now using a different accept content type to make the `/api/histories/{id}/exports` compatible with both export modes, it feels cleaner and works great with the interactive API docs :)